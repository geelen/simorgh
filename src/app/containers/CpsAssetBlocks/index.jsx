import React from 'react';
import pathOr from 'ramda/src/pathOr';
import { string, shape, arrayOf } from 'prop-types';

import HeadingBlock from './Blocks/Heading';
import LiveRadioBlock from './Blocks/LiveRadio';
import ParagraphBlock from './Blocks/Paragraph';
import UnknownBlock from './Blocks/Unknown';

const blockMap = {
  heading: HeadingBlock,
  paragraph: ParagraphBlock,
  liveradio: LiveRadioBlock,
};

const SKIP_LINK_ANCHOR_ID = 'content';

const CpsAssetBlocks = ({ blocks }) => {
  if (!blocks || !blocks.length) return null;

  return blocks.map((props, index) => {
    const { uuid, id, type } = props;
    const isFirstBlock = index === 0;
    const idAttr = isFirstBlock ? SKIP_LINK_ANCHOR_ID : null;
    const blockType = id || type;

    const Block = pathOr(UnknownBlock, [blockType], blockMap);

    return (
      <Block
        {...{
          ...props,
          idAttr,
          key: uuid,
          blockType,
        }}
      />
    );
  });
};

CpsAssetBlocks.propTypes = {
  blocks: arrayOf(
    shape({
      uuid: string,
      id: string,
      externalId: string,
      text: string,
      type: string,
    }),
  ).isRequired,
};

export default CpsAssetBlocks;
