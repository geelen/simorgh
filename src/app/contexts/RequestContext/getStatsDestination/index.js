/* Returns stats destnation for ATI based on origin, service and env
   see table on this issue https://github.com/bbc/simorgh/issues/2995
*/
const getStatsDestination = ({ isUK = true, env = 'test', service }) => {
  let destination = '';
  switch (service) {
    case 'news':
      // checks if "news" service is in the UK or not and set apprioprate destination
      destination = isUK !== false ? 'NEWS_PS' : 'NEWS_GNL';
      break;
    case 'cymrufyw':
    case 'naidheachdan':
      // checks if "cymrufyw" or "naidheachdan" service is in the UK or not and set apprioprate destination
      destination = isUK !== false ? 'NEWS_LANGUAGES_PS' : 'NEWS_LANGUAGES_GNL';
      break;
    case 'japanese':
      destination = 'NEWS_LANGUAGES_GNL';
      break;
    case 'scotland':
      destination = 'PLACEHOLDER';
      break;
    case 'archive':
      destination = 'BBC_ARCHIVE_PS';
      break;
    default:
      destination = 'WS_NEWS_LANGUAGES';
  }
  return env === 'live' ? destination : `${destination}_TEST`;
};

export default getStatsDestination;
