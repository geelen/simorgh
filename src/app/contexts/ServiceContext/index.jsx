import React from 'react';
import { node, string } from 'prop-types';
import services from '#lib/config/services/loadableConfig';
import { getVariant } from '../../lib/utilities/variantHandler';
import variantPropType from '../../models/propTypes/variants';

export const ServiceContext = React.createContext({});

export const ServiceContextProvider = ({ children, service, variant }) => {
  const LoadableContextProvider = services[service];

  if (!LoadableContextProvider) {
    return null;
  }

  return (
    <LoadableContextProvider
      Context={ServiceContext}
      dataKey={getVariant({ service, variant })}
    >
      {children}
    </LoadableContextProvider>
  );
};

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  service: string,
  variant: variantPropType,
};

ServiceContextProvider.defaultProps = {
  service: 'default',
  variant: 'default',
};
