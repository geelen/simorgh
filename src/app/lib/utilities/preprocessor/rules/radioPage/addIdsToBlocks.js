import pipe from 'ramda/src/pipe';
import path from 'ramda/src/path';
import map from 'ramda/src/map';
import mergeDeepLeft from 'ramda/src/mergeDeepLeft';
import uuid from 'uuid';

const addIdToBlock = block => ({ ...block, uuid: uuid() });

const getBlocks = path(['content', 'blocks']);

const mapBlocks = pipe(getBlocks, map(addIdToBlock));

export default jsonRaw =>
  mergeDeepLeft(
    {
      content: { blocks: mapBlocks(jsonRaw) },
    },
    jsonRaw,
  );
