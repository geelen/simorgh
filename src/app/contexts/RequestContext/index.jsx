import React from 'react';
import { bool, node, oneOf, string, number } from 'prop-types';
import getStatsDestination from './getStatsDestination';
import getStatsPageIdentifier from './getStatsPageIdentifier';
import getOriginContext from './getOriginContext';
import getEnv from './getEnv';
import getMetaUrls from './getMetaUrls';
import variantPropType from '../../models/propTypes/variants';

export const RequestContext = React.createContext({});

export const RequestContextProvider = ({
  children,
  bbcOrigin,
  id,
  isAmp,
  pageType,
  service,
  statusCode,
  previousPath,
  pathname,
  variant,
}) => {
  const { isUK, origin } = getOriginContext(bbcOrigin);
  const env = getEnv(origin);
  const platform = isAmp ? 'amp' : 'canonical';
  const statsDestination = getStatsDestination({
    isUK,
    env,
    service,
  });
  const statsPageIdentifier = getStatsPageIdentifier({
    pageType,
    service,
    id,
  });

  const value = {
    env,
    id,
    isUK,
    origin,
    pageType,
    isAmp,
    platform,
    statsDestination,
    statsPageIdentifier,
    statusCode,
    previousPath,
    variant,
    ...getMetaUrls(origin, pathname),
  };

  return (
    <RequestContext.Provider value={value}>{children}</RequestContext.Provider>
  );
};

RequestContextProvider.propTypes = {
  bbcOrigin: string,
  children: node.isRequired,
  id: string,
  isAmp: bool.isRequired,
  pageType: oneOf([
    'article',
    'frontPage',
    'media',
    'error',
    'MAP',
    'FIX',
    'STY',
    'PGL',
  ]).isRequired,
  service: string.isRequired,
  statusCode: number,
  pathname: string.isRequired,
  previousPath: string,
  variant: variantPropType,
};

RequestContextProvider.defaultProps = {
  bbcOrigin: null,
  id: null,
  statusCode: null,
  previousPath: null,
  variant: null,
};
