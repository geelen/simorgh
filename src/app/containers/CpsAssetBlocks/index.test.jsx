import React from 'react';
import path from 'ramda/src/path';
import { latin } from '@bbc/gel-foundations/scripts';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { ServiceContext } from '../../contexts/ServiceContext';
import { RequestContext } from '../../contexts/RequestContext';
import CpsAssetBlocks from '.';
import amharicPageData from '../../../../data/amharic/bbc_amharic_radio/liveradio';
import addIdsToBlocks from '../../routes/getInitialData/mediapage/addIdsToBlocks';

const pageData = addIdsToBlocks(amharicPageData);
const blocks = path(['content', 'blocks'], pageData);

describe('Cps Asset Blocks', () => {
  shouldMatchSnapshot(
    'should match snapshot',
    <ServiceContext.Provider value={{ service: 'news', script: latin }}>
      <RequestContext.Provider
        value={{ platform: 'canonical', pageType: 'media' }}
      >
        <CpsAssetBlocks blocks={blocks} />
      </RequestContext.Provider>
    </ServiceContext.Provider>,
  );
});
