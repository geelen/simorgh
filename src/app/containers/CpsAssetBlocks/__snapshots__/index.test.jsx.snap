// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media Page Blocks should match snapshot 1`] = `
.c0 {
  font-size: 1.75rem;
  line-height: 2rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  color: #3F3F42;
  margin: 0;
  padding: 2rem 0;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border: 0;
  overflow: hidden;
}

.c2 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c1 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  padding-bottom: 1.5rem;
  margin: 0;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c0 {
    font-size: 2rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    font-size: 2.75rem;
    line-height: 3rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    padding: 2.5rem 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c1 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

<div>
  <h1
    class="c0"
    id="content"
  >
    ያድምጡ
  </h1>
  <p
    class="c1"
  >
    ዝግጅቶቻችንን
  </p>
  <div
    class="c2"
  >
    <iframe
      allow="autoplay; fullscreen"
      allowfullscreen=""
      class="c3"
      scrolling="no"
      src="/ws/av-embeds/media/bbc_amharic_radio/liveradio"
    />
  </div>
</div>
`;
